searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_blueprint.cpp

```diff
@@ -51,24 +51,43 @@ struct ConvertCellsSelector
     }
 };
 
+vespalib::string
+to_string(NearestNeighborBlueprint::Algorithm algorithm)
+{
+    using NNBA = NearestNeighborBlueprint::Algorithm;
+    switch (algorithm) {
+        case NNBA::BRUTE_FORCE: return "brute force";
+        case NNBA::BRUTE_FORCE_FALLBACK: return "brute force fallback";
+        case NNBA::INDEX_TOP_K: return "index top k";
+        case NNBA::INDEX_TOP_K_WITH_FILTER: return "index top k using global filter";
+    }
+    return "unknown";
+}
+
 } // namespace <unnamed>
 
 NearestNeighborBlueprint::NearestNeighborBlueprint(const queryeval::FieldSpec& field,
                                                    const tensor::ITensorAttribute& attr_tensor,
                                                    std::unique_ptr<Value> query_tensor,
-                                                   uint32_t target_num_hits, bool approximate, uint32_t explore_additional_hits,
-                                                   double distance_threshold, double brute_force_limit)
+                                                   uint32_t target_num_hits,
+                                                   bool approximate,
+                                                   uint32_t explore_additional_hits,
+                                                   double distance_threshold,
+                                                   double global_filter_lower_limit,
+                                                   double global_filter_upper_limit)
     : ComplexLeafBlueprint(field),
       _attr_tensor(attr_tensor),
       _query_tensor(std::move(query_tensor)),
       _target_num_hits(target_num_hits),
       _approximate(approximate),
       _explore_additional_hits(explore_additional_hits),
       _distance_threshold(std::numeric_limits<double>::max()),
-      _brute_force_limit(brute_force_limit),
+      _global_filter_lower_limit(global_filter_lower_limit),
+      _global_filter_upper_limit(global_filter_upper_limit),
```

Comment:
`_global_filter_upper_limit` seems to be mostly unused (only used in `visitMembers`).